src/axl.c

```diff
@@ -69,11 +69,43 @@ int axl_copy_metadata;
  * the user is only going to be calling AXL_Create() a handful of times.  It
  * also simplifies the code if we never shrink it, and the extra memory usage
  * is negligible, if any at all.
+ *
+ * This structure is shared with the worker threads and must be protected using
+ * axl_kvtrees_obtain / axl_kvtrees_release.
  */
-kvtree** axl_kvtrees;
-static unsigned int axl_kvtrees_count = 0;
 
-static pthread_mutex_t id_lock = PTHREAD_MUTEX_INITIALIZER;
+/* lock and return all the AXL_Create'd kvtree pointers */
+static axl_kvtrees_t* axl_kvtrees_obtain(void)
```

Comment:
I would rename these functions to `axl_kvtrees_lock()`/`axl_kvtrees_unlock()` to match the pthreads conventions.

Reply:
That is actually the reason why I do not name them lock/unlock. They do more: they return a pointer to the object (obtain it).

Reply:
The latest version no longer protects `axl_kvtrees` and instead copies the required entry (the `file_list` pointer) when a new transfer is created. This removes the functions and any issue about their name.